examples/zero-nextjs/src/app/page.tsx

```diff
@@ -0,0 +1,213 @@
+  display: 'inline-block',
+  left: 'var(--x, 0)',
+  top: 'var(--y, 0)',
+  scale: '0',
+  translate: '-50% -50%',
+  mixBlendMode: 'multiply',
+  filter: 'blur(2px)',
+  animation: `${scale} var(--s, 2s) var(--d, 0s) infinite alternate`,
+});
+
+function randomBetween(min: number, max: number) {
+  return Math.floor(Math.random() * (max - min + 1) + min);
+}
+function generateBubbleVars() {
+  return `
+    --x: ${randomBetween(10, 90)}%;
+    --y: ${randomBetween(15, 85)}%;
+    --h: ${randomBetween(0, 360)};
+    --s2: ${randomBetween(2, 6)};
+    --d: -${randomBetween(250, 400) / 1000}s;
+    --s: ${randomBetween(3, 6)}s;
+  `;
+}
+
+export default function Home() {
+  return (
+    <main
+      className={css({
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        height: '100lvh',
+        padding: '20px',
+      })}
+    >
+      <h1
+        className={`
+        ${kalnia.className}
+        ${css`
+          font-size: clamp(3rem, 1.385rem + 6.8906vw, 8rem);
+          font-weight: 500;
+          line-height: 1;
+          text-align: center;
+          position: relative;
+          display: flex;
+          align-items: center;
+          color: #888;
+          margin-bottom: 1rem;
+
+          @media (prefers-color-scheme: dark) {
+            color: #fff;
+          }
+        `}`}
+      >
+        Zero Runtime
+        <span
+          className={css`
+            position: absolute;
+            inset: 0;
+            background: white;
+            mix-blend-mode: color-burn;
+            overflow: hidden;
+            pointer-events: none;
+
+            @media (prefers-color-scheme: dark) {
+              mix-blend-mode: darken;
+              filter: brightness(2);
+            }
+          `}
+        >
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
+            `}
+          />
+          <Bubble
+            className={css`
+              ${generateBubbleVars()}
```

Comment:
Just and FYI, These variables will be random during the build time. In the browser, it'll still be static.

Reply:
I think it's worth mentioning this behavior between emotion/zero-runtime:

```js
{[…Array(10)].map((_, index) => (
    <Bubble
      key={index}
      className={css`
        ${generateBubbleVars()}
      `
    />
)}
```

Zero-runtime: generate just one set of variables (all instances will have the same styles).
Emotion: generate 10 random set of styles.

Reply:
Emotion: generate 10 random set of styles that'll also change with each pave refresh.